ibc-core/ics02-client/types/src/events.rs

```diff
@@ -100,6 +166,37 @@ impl From<ConsensusHeightAttribute> for abci::EventAttribute {
     }
 }
 
+impl TryFrom<abci::EventAttribute> for ConsensusHeightAttribute {
+    type Error = ClientError;
+    fn try_from(value: abci::EventAttribute) -> Result<Self, Self::Error> {
+        value
+            .key_str()
+            .map(|key| {
+                if key != CONSENSUS_HEIGHT_ATTRIBUTE_KEY {
+                    return Err(ClientError::InvalidAttributeKey {
+                        attribute_key: key.to_string(),
+                    });
+                }
+                value
+                    .value_str()
+                    .map(|value| {
+                        Ok(ConsensusHeightAttribute {
+                            consensus_height: Height::from_str(value).map_err(|_| {
+                                ClientError::InvalidAttributeValue {
+                                    attribute_value: value.to_string(),
+                                }
+                            })?,
+                        })
+                    })
+                    .map_err(|_| ClientError::InvalidAttributeKey {
+                        attribute_key: String::new(),
+                    })?
+            })
+            .map_err(|_| ClientError::InvalidAttributeKey {
+                attribute_key: String::new(),
+            })?
+    }
+}
```

Comment:
```suggestion
impl TryFrom<abci::EventAttribute> for ConsensusHeightAttribute {
    type Error = ClientError;

    fn try_from(value: abci::EventAttribute) -> Result<Self, Self::Error> {
        if let Ok(key_str) = value.key_str() {
            if key_str != CONSENSUS_HEIGHT_ATTRIBUTE_KEY {
                return Err(ClientError::InvalidAttributeKey {
                    attribute_key: key_str.to_string(),
                });
            }
        } else {
            return Err(ClientError::InvalidAttributeKey {
                attribute_key: String::new(),
            });
        }

        value
            .value_str()
            .map(|val_str| {
                let consensus_height =
                    Height::from_str(val_str).map_err(|_| ClientError::InvalidAttributeValue {
                        attribute_value: val_str.to_string(),
                    })?;

                Ok(ConsensusHeightAttribute { consensus_height })
            })
            .map_err(|_| ClientError::InvalidAttributeKey {
                attribute_key: String::new(),
            })?
    }
}
```

Reply:
sorry I forgot to push the commit @seanchen1991